packages/catalog-process/src/services/catalogService.ts

```diff
@@ -3083,14 +3088,133 @@ export function catalogServiceBuilder(
           eserviceInstanceInterfaceData,
           config.eserviceTemplateDocumentsContainer,
           fileManager,
-          readModelService,
           ctx
         );
 
       await repository.createEvent(addDocumentEvent);
 
       return updatedEService;
     },
+    async createInstanceDescriptor(
+      eserviceId: EServiceId,
+      eserviceInstanceDescriptorSeed: catalogApi.EServiceInstanceDescriptorSeed,
+      ctx: WithLogger<AppContext>
+    ): Promise<Descriptor> {
+      ctx.logger.info(
+        `Creating Instance Descriptor for EService ${eserviceId}`
+      );
+
+      const eservice = await retrieveEService(eserviceId, readModelService);
+
+      await assertRequesterIsDelegateProducerOrProducer(
+        eservice.data.producerId,
+        eservice.data.id,
+        ctx.authData,
+        readModelService
+      );
+
+      assertHasNoDraftOrWaitingForApprovalDescriptor(eservice.data);
+      assertEServiceIsTemplateInstance(eservice.data);
+
+      const template = await retrieveEServiceTemplate(
+        eservice.data.templateRef.id,
+        readModelService
+      );
+
+      const latestDescriptor = getLatestDescriptor(eservice.data);
+      const templateVersion = template.versions.find(
+        (v) => v.id === latestDescriptor?.templateVersionRef?.id
+      );
+
+      if (!templateVersion) {
+        throw templateVersionNotFoundInEService(
+          eservice.data.templateRef.id,
+          eservice.data.id
+        );
+      }
+
+      const agreementApprovalPolicySeed =
+        eserviceInstanceDescriptorSeed.agreementApprovalPolicy
+          ? apiAgreementApprovalPolicyToAgreementApprovalPolicy(
+              eserviceInstanceDescriptorSeed.agreementApprovalPolicy
+            )
+          : undefined;
+
+      assertConsistentDailyCalls(eserviceInstanceDescriptorSeed);
+
+      const newDescriptor: Descriptor = createNextDescriptor(eservice.data, {
+        description: templateVersion.description,
+        voucherLifespan: templateVersion.voucherLifespan,
+        audience: eserviceInstanceDescriptorSeed.audience,
+        dailyCallsPerConsumer:
+          eserviceInstanceDescriptorSeed.dailyCallsPerConsumer,
+        dailyCallsTotal: eserviceInstanceDescriptorSeed.dailyCallsTotal,
```

Comment:
Double-check it's ok to ignore the values of `dailyCallsPerConsumer` and `dailyCallsTotal` from the template version. Reading the code I find confusing the fact that we instance something from a template but we ignore some data from the template itself...

For the approval policy, for example, if it's not in the seed (where it's optional) you get the one from the template.

```suggestion
        dailyCallsPerConsumer:
          eserviceInstanceDescriptorSeed.dailyCallsPerConsumer,
        dailyCallsTotal: eserviceInstanceDescriptorSeed.dailyCallsTotal,
```

Reply:
This might be a naming issue rather than logic.
From the SRS: the template's `dailyCallsPerConsumer` and `dailyCallsTotal` are treated from the template are treated as "default values" for the user that instanciate it